Documentation/project-docs/glossary.md

```diff
@@ -1,82 +1,227 @@
+* **IL**. Intermediate language. Higher level .NET languages, such as C#,
+  compile down to a hardware agnostic instruction set, which is called
+  Intermediate Language (IL). IL is sometimes referred to as MSIL (Microsoft IL)
+  or CIL (Common IL).
+
+* **GC**. Garbage collector. Garbage collection is an implementation of
+  automatic memory management. .NET Framework and .NET Core currently uses a
+  generational garbage collector, i.e. it groups objects into generations to
+  limit the number of nodes it has to walk for determining which objects are
+  alive. This speeds up collection times.
+
+* **JIT**. Just in time compiler. This technology compiles IL to machine code
+  that the processor understands. It's called JIT because compilation happens on
+  demand and is performed on the same machine the code needs to run on. Since
+  JIT compilation occurs during execution of the application, compile time is
+  part of the run time. Thus, JIT compilers have to trade spending more time
+  optimizing code with the savings the resulting code can produce. But a JIT
+  knows the actual hardware and can free developers from having to ship
+  different implementations. For instance, our vector library relies on the JIT
+  to use the highest available SIMD instruction set.
+
+* **AOT**. Ahead of time compiler. Similar to JIT, this compiler also translates
+  IL to machine code. In contrast to JIT compilation, AOT compilation happens
+  before the application is executed and is usually performed on a different
+  machine. AOT tool chains don't trade runtime for compile time and thus can
+  spend more time optimizing. Since the context of AOT is the entire
+  application, the AOT compiler can also perform cross module linking and whole
+  program analysis, which means that all references are followed and a single
+  executable is produced.
+
+* **NGEN**. Native (image) generation. You can think of this technology as a
+  persistent JIT compiler. It usually compiles code on the machine where the
+  code will be executed, but compilation typically occurs at install time.
+
+* **CoreFX**. Core framework. Conceptually a set of `System.*` (and to a limited
+  extent `Microsoft.*`) libraries that make up the lower layer of the .NET
+  library stack. It's what most people would think of as the Base Class Library
+  (BCL). The BCL is a general purpose, lower level set of functionality that
```

Comment:
Actually it's now more than the BCL (if the BCL concretely means, the "shared framework" otherwise known as Microsoft.NetCore.App, which is the set of types automatically referenced by all .NET Core apps). For example, CoreFX also contains the types in the Windows Compatibility pack.